src/blog/2023/09/making-sense-of-your-industrial-data.md

```diff
@@ -0,0 +1,87 @@
+description: Working with legacy industrial protocol data from the likes of Modbus and older, non IIoT protocols and putting it to work in an IIoT world.
+date: 2023-09-11
+authors: ["stephen-mclaughlin"]
+image: "/blog/2023/09/images/industrial-legacy-data-blog-image.png"
+tags:
+    - posts
+    - node-red
+    - community
+---
+
+In today's rapidly evolving industrial landscape, the ability to collect, analyse, and act upon data is more critical than ever.
+The Industrial Internet of Things (IIoT) has ushered in a new era of connectivity and automation, but what about the wealth of data locked away in legacy industrial protocols like ModBus?
+How can we unlock and make sense of this valuable information in the IIoT era? That's where tools like Node-RED and `node-red-contrib-buffer-parser` come into play.
+
+<!--more-->
+
+### Legacy Industrial Data: ModBus
+
+ModBus is one of the most widely used industrial protocols. Originally developed in the late 1970s, it has been a popular choice for industrial communication ever since. However, its data format can be challenging to work with in the context of modern IIoT applications. ModBus typically represents data in 16-bit unsigned registers, making it necessary to convert this data into more usable formats like Signed integer, Float, Signed and Unsigned Long, String, or even individual bits.
+
+### A short primer on data types
+
+Before we dive into how to make sense of ModBus data, let's take a quick look at some of the common data types we have to deal with.
+
+#### 16-bit unsigned
+
+16-bit unsigned data is an integer that can only be positive. It can represent values from 0 to 65535. For example, the number 12345 is represented as `0x3039` in hexadecimal or `0011000000111001` in binary.
+
+#### 16-bit signed
+
+16-bit signed data is an integer that can be positive or negative. It can represent values from -32768 to 32767. For example, the number -12345 is represented as `0xCFC7` in hexadecimal or `1100111111000111` in binary.
+
+#### 32-bit data
+
+32-bit data, like 16-bit data can mean many things. It could be signed or unsigned, or even a floating point number. For example, the number 12345 is represented as `0x00003039` in hexadecimal or `00000000000000000011000000111001` in binary. Typically, 32-bit data is represented as two 16-bit registers. Therefore, when dealing with 32-bit data, you need to combine two 16-bit registers to get the full value.
+
+#### Endianness
+
+Endianness is a term used to describe the order in which bytes are stored in memory. There are two types of endianness: big-endian and little-endian. In big-endian, the most significant byte is stored first, while in little-endian, the least significant byte is stored first. For example, the number 12345 is represented as `0x3039` in big-endian and `0x3930` in little-endian.  In other words, the bytes might be stored in reverse order. This further complicates the process of converting ModBus data into more usable formats.
```

Comment:
```suggestion
Endianness, particularly in the context data communications, refers to the order or the sequence of how bytes are stored or transmitted. There are two types of endianness: big-endian (BE) and little-endian (LE). In big-endian, the most significant byte is first, while in little-endian, the least significant byte is first. For example, the number 12345 is represented as `0x3039` in a big-endian word and `0x3930` in little-endian word.  This can often cause confusion and complicate the process of converting ModBus data into more usable formats.
```

Perhaps this suggestion removes the confusion a little.  I dont want to fully remove confusion (😉) or even remove this section.  I feel it has value for demonstrating that "we" (Node-red/buffer parser) can help you make sense of the confusion and difficulties legacy protocols can present (and partly why the very next sentence says "Node-RED and node-red-contrib-buffer-parser to the rescue")